prov/efa/src/rxr/rxr_cq.c

```diff
@@ -475,45 +476,46 @@ void rxr_cq_complete_rx(struct rxr_ep *ep,
 	 * action of sending ctrl packet may cause the release of RX entry (when inject
 	 * was used on lower device).
 	 */
-	if (rx_entry->cq_entry.flags & FI_WRITE) {
+	if (op_entry->cq_entry.flags & FI_WRITE) {
 		/*
 		 * For write, only write RX completion when REMOTE_CQ_DATA is on
 		 */
-		if (rx_entry->cq_entry.flags & FI_REMOTE_CQ_DATA)
-			rxr_cq_write_rx_completion(ep, rx_entry);
-	} else if (rx_entry->cq_entry.flags & FI_READ) {
-		/* This rx_entry is part of the for emulated read protocol,
+		if (op_entry->cq_entry.flags & FI_REMOTE_CQ_DATA)
+			rxr_cq_write_rx_completion(ep, op_entry);
```

Comment:
should we rename `rxr_cq_write_rx_completion` to `rxr_cq_write_recv_completion`?

Reply:
You are right, I missed that. Makes sense.